extensions/scheduler/common/src/main/java/io/quarkus/scheduler/common/runtime/AbstractJobDefinition.java

```diff
@@ -0,0 +1,91 @@
+package io.quarkus.scheduler.common.runtime;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import io.quarkus.scheduler.Scheduled.ConcurrentExecution;
+import io.quarkus.scheduler.Scheduled.SkipPredicate;
+import io.quarkus.scheduler.ScheduledExecution;
+import io.quarkus.scheduler.Scheduler.JobDefinition;
+import io.smallrye.mutiny.Uni;
+
+public abstract class AbstractJobDefinition implements JobDefinition {
+
+    protected final String identity;
+    protected String cron = "";
+    protected String every = "";
+    protected String delayed = "";
+    protected String overdueGracePeriod = "";
+    protected ConcurrentExecution concurrentExecution = ConcurrentExecution.PROCEED;
+    protected SkipPredicate skipPredicate = null;
+    protected Consumer<ScheduledExecution> task;
+    protected Function<ScheduledExecution, Uni<Void>> asyncTask;
+    protected boolean scheduled = false;
+
+    public AbstractJobDefinition(String identity) {
+        this.identity = identity;
+    }
+
+    @Override
+    public JobDefinition setCron(String cron) {
+        checkScheduled();
+        this.cron = cron;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setInterval(String every) {
+        checkScheduled();
+        this.every = every;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setDelayed(String period) {
+        checkScheduled();
+        this.delayed = period;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setConcurrentExecution(ConcurrentExecution concurrentExecution) {
+        checkScheduled();
+        this.concurrentExecution = concurrentExecution;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setSkipPredicate(SkipPredicate skipPredicate) {
+        checkScheduled();
+        this.skipPredicate = skipPredicate;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setOverdueGracePeriod(String period) {
+        checkScheduled();
+        this.overdueGracePeriod = period;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setTask(Consumer<ScheduledExecution> task) {
+        checkScheduled();
+        this.task = task;
+        return this;
+    }
+
+    @Override
+    public JobDefinition setAsyncTask(Function<ScheduledExecution, Uni<Void>> asyncTask) {
+        checkScheduled();
```

Comment:
is it possible to fail fast here if task was already defined ? (and not having to wait until schedule() is called to get the error)

Reply:
:+1: Yep, that makes sense.